flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -80,14 +83,17 @@ public FlinkCatalog(
       String catalogName,
       String defaultDatabase,
       String[] baseNamespace,
-      Catalog icebergCatalog,
+      CatalogLoader catalogLoader,
+      Configuration hadoopConf,
       boolean cacheEnabled) {
     super(catalogName, defaultDatabase);
-    this.originalCatalog = icebergCatalog;
-    this.icebergCatalog = cacheEnabled ? CachingCatalog.wrap(icebergCatalog) : icebergCatalog;
+    this.hadoopConf = hadoopConf;
+    this.originalCatalog = catalogLoader.loadCatalog(hadoopConf);
```

Comment:
Is possible to track only on `Catalog` in this `FlinkCatalog` class ? For example,  we only keep the `icebergCatalog` as the member of this class, that will be much easier to follow the code ( Introducing two member here confused me sometime).   When `close` the catalog we could make the `CachingCatalog` to implement `Closeable` inteface.

Reply:
I think we can.

Reply:
Maybe we can let `Catalog` extends `Closable`.

Reply:
`CachingCatalog` is just a wrapper for original catalog. I think it is better to not let it implement Closeable interface.
A way to solve this problem can be: Keep `Closeable` as a class member instead of `Catalog`.
`closeable = originalCatalog instanceof Closeable ? (Closeable) originalCatalog : null;`
What do you think?

Reply:
Make sense.